src/AddressRegistry.Producer.Ldes/Infrastructure/Program.cs

```diff
@@ -0,0 +1,170 @@
+
+            AppDomain.CurrentDomain.UnhandledException += (_, eventArgs) =>
+                Log.Fatal((Exception)eventArgs.ExceptionObject, "Encountered a fatal exception, exiting program.");
+
+            Log.Information("Initializing AddressRegistry.Producer.Ldes");
+
+            var host = new HostBuilder()
+                .ConfigureAppConfiguration((_, configurationBuilder) =>
+                {
+                    configurationBuilder
+                        .SetBasePath(Directory.GetCurrentDirectory())
+                        .AddJsonFile("appsettings.json", optional: false, reloadOnChange: false)
+                        .AddJsonFile($"appsettings.{Environment.MachineName.ToLowerInvariant()}.json", optional: true, reloadOnChange: false)
+                        .AddEnvironmentVariables()
+                        .AddCommandLine(args);
+                })
+                .ConfigureLogging((hostContext, loggingBuilder) =>
+                {
+                    SelfLog.Enable(Console.WriteLine);
+                    Log.Logger = new LoggerConfiguration() //NOSONAR logging configuration is safe
+                        .ReadFrom.Configuration(hostContext.Configuration)
+                        .Enrich.FromLogContext()
+                        .Enrich.WithMachineName()
+                        .Enrich.WithThreadId()
+                        .Enrich.WithEnvironmentUserName()
+                        .Destructure.JsonNetTypes()
+                        .CreateLogger();
+                    loggingBuilder.ClearProviders();
+                    loggingBuilder.AddSerilog(Log.Logger);
+                })
+                .ConfigureServices((hostContext, services) =>
+                {
+                    var healthChecksBuilder = services.AddHealthChecks();
+                    var connectionStrings = hostContext.Configuration
+                        .GetSection("ConnectionStrings")
+                        .GetChildren();
+
+                    foreach (var connectionString in connectionStrings
+                                 .Where(x => !x.Value.Contains("host", StringComparison.OrdinalIgnoreCase)))
+                    {
+                        healthChecksBuilder.AddSqlServer(
+                            connectionString.Value,
+                            name: $"sqlserver-{connectionString.Key.ToLowerInvariant()}",
+                            tags: new[] { "db", "sql", "sqlserver" });
+                    }
+
+                    foreach (var connectionString in connectionStrings
+                                 .Where(x => x.Value.Contains("host", StringComparison.OrdinalIgnoreCase)))
+                    {
+                        healthChecksBuilder.AddNpgSql(
+                            connectionString.Value,
+                            name: $"npgsql-{connectionString.Key.ToLowerInvariant()}",
+                            tags: new[] { "db", "sql", "npgsql" });
+                    }
```

Comment:
npgsql is not needed, also remove reference

Reply:
done